### PR TITLE
Adding QE Intern Update blog #2 in pulpproject.org

### PR DESCRIPTION
**Pull request by ragabala**
This is a blog stating my experience as a QE intern in the pulp
project(update-2).
Creating this request for merging it with the pulp project upstream.
This blog is meant to be displayed under the blogs tab in the
pulpproject.org.

refer
https://ragabala.github.io/pulpproject.org/2018/06/22/pulp_qe_intern_update/

### CODE REVIEW

File: _posts/2018-06-22-pulp_qe_intern_update.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+---
+title: 2018 Pulp QE Intern Update #2
+author: Ragavendran Balakrishnan
+tags:
+  - pulpqe
+---
+
+## Journey so Far
+
+The first two weeks of my Red Hat journey was a roller coaster ride with lots of take in's, which I
```
Comment (daviddavis) [inline _posts/2018-06-22-pulp_qe_intern_update.md]: A "take-in" is a deception or con. I don't think "take-in" is the right word here. Maybe "takeaways"? Or lessons, learning experiences, etc?

File: _posts/2018-06-22-pulp_qe_intern_update.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+have covered in my [first blog](https://pulpproject.org/2018/06/08/pulp_qe_intern_update/). Now,
+Here I am in the third week of my Internship and I have started writing my second blog post  on my
```
Comment (daviddavis) [inline _posts/2018-06-22-pulp_qe_intern_update.md]: nitpick - two spaces after post. Also "Here" and "Internship" should be lowercase.

File: _posts/2018-06-22-pulp_qe_intern_update.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+## FIPS
+
+FIPS stands for Federal Information Processing Standards. As the name suggests, it has something to
+do with the federal government in the United States (duh!). Basically, it provides a set of standards,
+which any machine that wants to be FIPS compliant, has to follow. These standards establish system
+security and interoperability.
+
+As one of the core goals in my internship is to install and test Pulp in a FIPS enabled RHEL
+machine, I started gathering information on enabling FIPS on VMs. Then I enabled FIPS in a Cent OS
```
Comment (daviddavis) [inline _posts/2018-06-22-pulp_qe_intern_update.md]: s/Cent OS/CentOS/

File: _posts/2018-06-22-pulp_qe_intern_update.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+machine. Have documented the steps and hiccups that I faced in an internal blog as well. Once I got
+it setup in CentOS, I moved on to check the same in a RHEL VM, with an "Employee subscription" and
```
Comment (daviddavis) [inline _posts/2018-06-22-pulp_qe_intern_update.md]: Unnecessary comma after "VM"

File: _posts/2018-06-22-pulp_qe_intern_update.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+it worked like a charm.
+
+## Running Pulp Smash on FIPS enabled machine
+
+Once, I got a FIPS enabled RHEL machine (ran it in a Virtual Machine), the next step was installing
```
Comment (daviddavis) [inline _posts/2018-06-22-pulp_qe_intern_update.md]: Drop the comma after "Once"

File: _posts/2018-06-22-pulp_qe_intern_update.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+Pulp in it by running an [Ansible Playbook](https://github.com/pulp/pulp-ci). This, however, threw
+some unexpected errors and it got resolved by using PULP 2.17 nightly build. After this, it was time
+for me to test, which I carried out by running pulp-smash against the aforementioned machine. Again
+I faced some issues, which took some time for me to figure out that it was happening due to the SSL
+connections with the VM. Finally, I decided to run pulp-smash locally on the VM where Pulp is
+installed, and it worked.
+
+The actual testing is done in a "compare and report" fashion, Where I had another pulp-installed-
```
Comment (daviddavis) [inline _posts/2018-06-22-pulp_qe_intern_update.md]: s/Where/where/

File: _posts/2018-06-22-pulp_qe_intern_update.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+non-FIPS-RHEL-VM. I made sure, it had all the configurations of its FIPS counterpart(of course
```
Comment (daviddavis) [inline _posts/2018-06-22-pulp_qe_intern_update.md]: s/counterpart(of/counterpart (of/

File: _posts/2018-06-22-pulp_qe_intern_update.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+except the fips switch being turned off). Then using the power of Tmux synchronize panes, I was able
```
Comment (daviddavis) [inline _posts/2018-06-22-pulp_qe_intern_update.md]: s/fips/FIPS/

Comment (daviddavis) [reply _posts/2018-06-22-pulp_qe_intern_update.md]: s/Tmux/tmux/

File: _posts/2018-06-22-pulp_qe_intern_update.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+to execute tests simultaneously on the two VMs (one with FIPS enabled and the other without FIPS)
+and drew conclusions based on the logs generated by my test runner (Used Pytests for running those
+tests). Based on my observations, I even raised a couple of bugs (cheers!!!).
+
+One of the things which helped me in analyzing the issue is using the python debugger, that allowed
+me to set breakpoints in the test execution and analyze step by step. some trace of using the
+debugger is briefed in the [bug](https://pulp.plan.io/issues/3769), that I raised.
```
Comment (daviddavis) [inline _posts/2018-06-22-pulp_qe_intern_update.md]: 👍

File: _posts/2018-06-22-pulp_qe_intern_update.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+##  Jenkins and Ansible
+
+One of the coolest things that I got to do in these two weeks was learning extensively about the CI
+used in the pulp project. I enrolled myself in the Ansible certification course in Redhat University
```
Comment (daviddavis) [inline _posts/2018-06-22-pulp_qe_intern_update.md]: s/pulp/Pulp/

Comment (daviddavis) [reply _posts/2018-06-22-pulp_qe_intern_update.md]: s/Redhat/Red Hat/

**Comment by daviddavis** [comment]
Great job 👍

**Comment by ragabala** [comment]
Hi @daviddavis,

Have made the required changes. Please do let me know if this can be merged.

Thanks!!

File: _posts/2018-06-22-pulp_qe_intern_update.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+machine, I started gathering information on enabling FIPS on VMs. Then I enabled FIPS in a CentOS
+machine. Have documented the steps and hiccups that I faced in an internal blog as well. Once I got
+it setup in CentOS, I moved on to check the same in a RHEL VM with an "Employee subscription" and
+it worked like a charm.
+
+## Running Pulp Smash on FIPS enabled machine
+
+Once I got a FIPS enabled RHEL machine (ran it in a Virtual Machine), the next step was installing
+Pulp in it by running an [Ansible Playbook](https://github.com/pulp/pulp-ci). This, however, threw
+some unexpected errors and it got resolved by using PULP 2.17 nightly build. After this, it was time
```
Comment (daviddavis) [inline _posts/2018-06-22-pulp_qe_intern_update.md]: This should be Pulp. Sorry I didn't catch it before.

**Comment by daviddavis** [comment]
One small comment. Mind fixing it and bumping the date to today (June 28)?

Then I'll merge. Thanks @ragabala!

**Comment by ragabala** [comment]
@daviddavis Thanks Once again. Have made the required changes.

**Comment by daviddavis** [comment]
Thanks @ragabala!

**Pull request merged**